src/audio/dai-zephyr.c

```diff
@@ -308,8 +307,8 @@ dai_dma_cb(struct dai_data *dd, struct comp_dev *dev, uint32_t bytes,
 				}
 
 				if (sink_dev && sink_dev->state == COMP_STATE_ACTIVE)
-					ret = dma_buffer_copy_from_no_consume(dma_buf, sink_c,
-									      converter[j], bytes);
+					dma_buffer_copy_from_no_consume(dma_buf, sink_c,
+									converter[j], bytes);
 err:
```

Comment:
not related to this your PR and probably not relevant for performance, but could we make the code above more readable and less error prone by removing those mid-function jumps and rewriting it as
```
	if (j >= IPC4_COPIER_MODULE_OUTPUT_PINS_COUNT) {
		comp_err(dev, "Sink queue ID: %d >= max output pin count: %d\n",
			 j, IPC4_COPIER_MODULE_OUTPUT_PINS_COUNT);
		ret = -EINVAL;
	} else if (!converter[j]) {
		comp_err(dev, "No PCM converter for sink queue %d\n", j);
		ret = -EINVAL;
	} else if (sink_dev && sink_dev->state == COMP_STATE_ACTIVE) {
		ret = dma_buffer_copy_from_no_consume(dma_buf, sink_c,
						      converter[j], bytes);
	}
```

Reply:
Those code are added may be can resolve some CI or QB failures, @ranj063 may comment on this.

Actually not only this, there is a lot of code, like below in dai_common_copy:

	/* Check possibility of glitch occurrence */
	if (dev->direction == SOF_IPC_STREAM_PLAYBACK &&
	    copy_bytes + avail_bytes < dd->period_bytes)
		comp_warn(dev, "dai_common_copy(): Copy_bytes %d + avail bytes %d < period bytes %d, possible glitch",
			  copy_bytes, avail_bytes, dd->period_bytes);
	else if (dev->direction == SOF_IPC_STREAM_CAPTURE &&
		 copy_bytes + free_bytes < dd->period_bytes)
		comp_warn(dev, "dai_common_copy(): Copy_bytes %d + free bytes %d < period bytes %d, possible glitch",
			  copy_bytes, free_bytes, dd->period_bytes);

	/* return if nothing to copy */
	if (!copy_bytes) {
		comp_warn(dev, "dai_zephyr_copy(): nothing to copy");
		dma_reload(dd->chan->dma->z_dev, dd->chan->index, 0, 0, 0);
		return 0;
	}

I will try to remove or optimize those in next round of copier optimization.

Reply:
@btian1 this is code carried over from the previous implementation when we did the optimization or conversion to module interface. If there's scope for clean up/further optimization lets do it.